src/server/readpool/mod.rs

```diff
@@ -25,49 +24,64 @@ use util;
 use util::futurepool::{self, FuturePool};
 
 pub use self::config::Config;
-pub use self::context::Context;
 pub use self::priority::Priority;
 
 const TICK_INTERVAL_SEC: u64 = 1;
 
-#[derive(Clone)]
-pub struct ReadPool {
-    pool_high: FuturePool<Context>,
-    pool_normal: FuturePool<Context>,
-    pool_low: FuturePool<Context>,
+pub struct ReadPool<T: futurepool::Context + 'static> {
+    pool_high: FuturePool<T>,
+    pool_normal: FuturePool<T>,
+    pool_low: FuturePool<T>,
     max_tasks_high: usize,
     max_tasks_normal: usize,
     max_tasks_low: usize,
 }
 
-impl util::AssertSend for ReadPool {}
-impl util::AssertSync for ReadPool {}
+impl<T: futurepool::Context + 'static> util::AssertSend for ReadPool<T> {}
+impl<T: futurepool::Context + 'static> util::AssertSync for ReadPool<T> {}
 
-impl ReadPool {
-    pub fn new(config: &Config) -> Self {
+impl<T: futurepool::Context + 'static> Clone for ReadPool<T> {
+    fn clone(&self) -> Self {
+        ReadPool {
+            pool_high: self.pool_high.clone(),
+            pool_normal: self.pool_normal.clone(),
+            pool_low: self.pool_low.clone(),
+            ..*self
+        }
+    }
+}
+
+impl<T: futurepool::Context + 'static> ReadPool<T> {
+    // Rust does not support copying closures (RFC 2132) so that we need a closure builder.
+    // TODO: Use a single closure once RFC 2132 is implemented.
```

Comment:
Use `Arc<Fn()>` is also avaliable. You can take a look at `CpuPool::after_start` later.